src/trajectory.jl

```diff
@@ -149,22 +151,25 @@ function transition(
     h::Hamiltonian,
     z::PhasePoint
 ) where {T<:Real}
+    H0 = energy(z)
     z′ = step(rng, τ.integrator, h, z, τ.n_steps)
     # Accept via MH criteria
-    is_accept, α = mh_accept_ratio(rng, energy(z), energy(z′))
+    is_accept, α = mh_accept_ratio(rng, H0, energy(z′))
```

Comment:
Why do we change this again? Seems unnecessary.

Reply:
We need H0 down below to compute `hamiltonian_energy_error`, so we cache it above and reuse it here.

Reply:
Oh OK I see you intention. It's OK to keep as it but actually `energy(z)` caches the computation already as the energy is computed when `z` is constructed.

Reply:
Got it. I figured just for generality it would be nice to keep it this way so that this function isn't dependent on `energy`'s implementation details.